[CI] DataStreamIndexSettingsProviderTests testGetAdditionalIndexSettingsMappingsMerging failing

**Build Scans:**
- [elasticsearch-periodic #11032 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/k2rlaypmp4yiq)
- [elasticsearch-intake #29506 / part1](https://gradle-enterprise.elastic.co/s/zxvjr2ho4gvzw)
- [elasticsearch-periodic #10969 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/5nlyayg2subvc)
- [elasticsearch-periodic #10962 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/rvrojuvyxnlec)

**Reproduction Line:**
```
./gradlew ":modules:data-streams:test" --tests "org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests.testGetAdditionalIndexSettingsMappingsMerging" -Dtests.seed=8921766E7DA7DE06 -Dtests.locale=xog-Latn-UG -Dtests.timezone=America/Indiana/Tell_City -Druntime.java=25
```

**Applicable branches:**
9.2

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testGetAdditionalIndexSettingsMappingsMerging'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: iterable with items ["field1", "field3"] in any order
     but: no item matches: "field1", "field3" in []
```

**Issue Reasons:**
- [9.2] 4 failures in test testGetAdditionalIndexSettingsMappingsMerging (1.6% fail rate in 252 executions)
- [9.2] 3 failures in pipeline elasticsearch-periodic (27.3% fail rate in 11 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProviderTests.java
+++ b/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProviderTests.java
@@ -71,7 +71,9 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
         String dataStreamName = "logs-app1";
 
         Instant now = Instant.now().truncatedTo(ChronoUnit.SECONDS);
-        Settings settings = Settings.EMPTY;
+        Settings settings = Settings.builder()
+            .put("index.dimensions_tsid_strategy_enabled", indexDimensionsTsidStrategyEnabledSetting)
+            .build();
         String mapping = """
             {
                 "_doc": {
@@ -117,11 +119,8 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
         Settings result = additionalSettings.build();
         // The index.time_series.end_time setting requires index.mode to be set to time_series adding it here so that we read this setting:
         // (in production the index.mode setting is usually provided in an index or component template)
-        result = builder().put(result)
-            .put("index.mode", "time_series")
-            .put("index.dimensions_tsid_strategy_enabled", indexDimensionsTsidStrategyEnabledSetting)
-            .build();
-        assertThat(result.size(), equalTo(5));
+        result = builder().put(result).put("index.mode", "time_series").build();
+        assertThat(result.size(), equalTo(4));
         assertThat(IndexSettings.MODE.get(result), equalTo(IndexMode.TIME_SERIES));
         assertThat(IndexSettings.TIME_SERIES_START_TIME.get(result), equalTo(now.minusMillis(DEFAULT_LOOK_BACK_TIME.getMillis())));
         assertThat(IndexSettings.TIME_SERIES_END_TIME.get(result), equalTo(now.plusMillis(DEFAULT_LOOK_AHEAD_TIME.getMillis())));
@@ -187,7 +186,9 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
         String dataStreamName = "logs-app1";
 
         Instant now = Instant.now().truncatedTo(ChronoUnit.SECONDS);
-        Settings settings = Settings.EMPTY;
+        Settings settings = Settings.builder()
+            .put("index.dimensions_tsid_strategy_enabled", indexDimensionsTsidStrategyEnabledSetting)
+            .build();
         String mapping1 = """
             {
                 "_doc": {
@@ -251,9 +252,8 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
         // (in production the index.mode setting is usually provided in an index or component template)
         result = builder().put(result)
             .put("index.mode", "time_series")
-            .put("index.dimensions_tsid_strategy_enabled", indexDimensionsTsidStrategyEnabledSetting)
             .build();
-        assertThat(result.size(), equalTo(5));
+        assertThat(result.size(), equalTo(4));
         assertThat(IndexSettings.MODE.get(result), equalTo(IndexMode.TIME_SERIES));
         assertThat(IndexSettings.TIME_SERIES_START_TIME.get(result), equalTo(now.minusMillis(DEFAULT_LOOK_BACK_TIME.getMillis())));
         assertThat(IndexSettings.TIME_SERIES_END_TIME.get(result), equalTo(now.plusMillis(DEFAULT_LOOK_AHEAD_TIME.getMillis())));

--- a/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProviderTests.java
+++ b/modules/data-streams/src/test/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProviderTests.java
@@ -250,9 +250,7 @@ public class DataStreamIndexSettingsProviderTests extends ESTestCase {
         Settings result = additionalSettings.build();
         // The index.time_series.end_time setting requires index.mode to be set to time_series adding it here so that we read this setting:
         // (in production the index.mode setting is usually provided in an index or component template)
-        result = builder().put(result)
-            .put("index.mode", "time_series")
-            .build();
+        result = builder().put(result).put("index.mode", "time_series").build();
         assertThat(result.size(), equalTo(4));
         assertThat(IndexSettings.MODE.get(result), equalTo(IndexMode.TIME_SERIES));
         assertThat(IndexSettings.TIME_SERIES_START_TIME.get(result), equalTo(now.minusMillis(DEFAULT_LOOK_BACK_TIME.getMillis())));

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -612,15 +612,9 @@ tests:
 - class: org.elasticsearch.xpack.esql.action.CrossClusterQueryWithPartialResultsIT
   method: testOneRemoteClusterPartial
   issue: https://github.com/elastic/elasticsearch/issues/124055
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettingsMappingsMerging
-  issue: https://github.com/elastic/elasticsearch/issues/135884
 - class: org.elasticsearch.xpack.esql.action.EsqlActionBreakerIT
   method: testTopNPushedToLuceneOnSortedIndex
   issue: https://github.com/elastic/elasticsearch/issues/135939
-- class: org.elasticsearch.datastreams.DataStreamIndexSettingsProviderTests
-  method: testGetAdditionalIndexSettings
-  issue: https://github.com/elastic/elasticsearch/issues/135972
 - class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
   method: test {yaml=indices.get_sample/10_basic/Test get sample for index with no sample config}
   issue: https://github.com/elastic/elasticsearch/issues/135975

